File: algorithims/sorting/quick_sort.py
def quick_sort(arr):
    if len(arr)<=1:
        return arr
    
    mid = arr[len(arr)//2]
    left = [x for x in arr if x < mid]
    middle = [x for x in arr if x == mid]
    right = [x for x in arr if x > mid]

    return quick_sort(left) + middle + quick_sort(right)

arr = [6,10,1,100,27,9]
print(quick_sort(arr))

File: algorithims/sorting/quickSort.js
var quickSort = (arr) => {
    if (arr.length <= 1) {
        return arr
    }

    let mid = arr[Math.floor(arr.length/2)]
    let left = []
    let right = []
    let middle = []

    for (x of arr) {
        if (x<mid) {
            left.push(x)
        }
        else if (x > mid) {
            right.push(x)
        }
        else{
            middle.push(x)
        }
    }

    return quickSort(left).concat(middle, quickSort(right))
}

arr = [6,10,1,100,27,9]
console.log(quickSort(arr))